packages/components/date-picker/src/date-picker-com/panel-date-range.ts

```diff
@@ -0,0 +1,190 @@
+  })
+
+  const rightYear = computed(() => {
+    return rightDate.value.year()
+  })
+  const rightMonth = computed(() => {
+    return rightDate.value.month()
+  })
+
+  const computedYearLabel = (
+    currentView: Ref<CurrentView>,
+    yearValue: ComputedRef<number>
+  ) => {
+    const yearTranslation = t('el.datepicker.year')
+    if (currentView.value === 'year') {
+      const startYear = Math.floor(yearValue.value! / 10) * 10
+      return yearTranslation
+        ? `${startYear} ${yearTranslation} - ${
+            startYear + 9
+          } ${yearTranslation}`
+        : `${startYear} - ${startYear + 9}`
+    }
+    return `${yearValue.value} ${yearTranslation}`
+  }
+
+  const focusPicker = (currentViewRef?: CurrentViewRef) => {
+    currentViewRef?.focus()
+  }
+
+  const showPicker = async (
+    pickerType: 'left' | 'right',
+    view: 'month' | 'year'
+  ) => {
+    const currentView =
+      pickerType === 'left' ? leftCurrentView : rightCurrentView
+    const currentViewRef =
+      pickerType === 'left' ? leftCurrentViewRef : rightCurrentViewRef
+    currentView.value = view
+    await nextTick()
+    focusPicker(currentViewRef.value)
+  }
+
+  const adjustDate = (
+    type: 'left' | 'right',
+    startDate: Ref<Dayjs>,
+    endDate: Ref<Dayjs>
+  ) => {
+    if (!props.unlinkPanels) {
+      endDate.value =
+        type === 'left'
+          ? startDate.value.add(1, 'month')
+          : startDate.value.subtract(1, 'month')
+    }
+  }
+
+  const yearPick = async (
+    pickerType: 'left' | 'right',
+    year: number | YearsPickerEmits
+  ) => {
+    const startDate = pickerType === 'left' ? leftDate : rightDate
+    const endDate = pickerType === 'left' ? rightDate : leftDate
+    const currentView =
+      pickerType === 'left' ? leftCurrentView : rightCurrentView
+    const currentViewRef =
+      pickerType === 'left' ? leftCurrentViewRef : rightCurrentViewRef
+
+    const data = startDate.value.year(year as number)
+    startDate.value = getValidDateOfYear(data, lang.value, disabledDate)
+    adjustDate(pickerType, startDate, endDate)
+
+    currentView.value = 'month'
+    await nextTick()
+    focusPicker(currentViewRef.value)
+    handlePanelChange('year')
+  }
+
+  const monthPick = async (
+    pickerType: 'left' | 'right',
+    month: number | MonthsPickerEmits
+  ) => {
+    const startDate = pickerType === 'left' ? leftDate : rightDate
+    const endDate = pickerType === 'left' ? rightDate : leftDate
+    const currentView =
+      pickerType === 'left' ? leftCurrentView : rightCurrentView
+    const currentViewRef =
+      pickerType === 'left' ? leftCurrentViewRef : rightCurrentViewRef
+    startDate.value = getValidDateOfMonth(
+      startDate.value.year(),
+      month as number,
+      lang.value,
+      disabledDate
+    )
+    adjustDate(pickerType, startDate, endDate)
+
+    currentView.value = 'date'
+    await nextTick()
+    focusPicker(currentViewRef.value)
+    handlePanelChange('year')
+  }
```

Comment:
The logic overlap between `monthPick` and `yearPick` is also quite high, they may be extracted.